sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/InteractiveBrowserCredential.java

```diff
@@ -44,7 +47,10 @@ public InteractiveBrowserCredential(int port, IdentityClientOptions identityClie
 
     @Override
     public Mono<AccessToken> getToken(String... scopes) {
-        validate();
-        return identityClient.authenticateWithBrowserInteraction(tenantId(), clientId(), scopes, port);
+        if (clientId() == null) {
+            throw new IllegalArgumentException("Must provide non-null value for client id in " + this.getClass().getSimpleName());
```

Comment:
Why do this check every time you call getToken.  Couldn't this be done up front when the credential is constructed?

Reply:
Let me talk to Jonathan about this one - currently the client id parameter is mutable.